### PR TITLE
pkg/steps/lease: add support for multiple leases

### PR DESCRIPTION
**Pull request by bbguimaraes**
An internal change that should not have any externally observable behavior
change (other than a few log messages).  Required to support multiple
leases per test in the future.

/hold

Reviewable, but I'll work on adding tests for the existing code tomorrow.

### CODE REVIEW

File: pkg/steps/lease.go [hunk 1]
```diff
@@ -73,25 +82,25 @@ func (s *leaseStep) Run(ctx context.Context) error {
...
+		}
+		return err
 	}
 	heartbeatCancel()
-	log.Printf("Acquired lease %q for %q", lease, s.leaseType)
-	s.leasedResource = lease
 	wrappedErr := results.ForReason("executing_test").ForError(s.wrapped.Run(ctx))
-	log.Printf("Releasing lease for %q", s.leaseType)
-	releaseErr := results.ForReason("releasing_lease").ForError(client.Release(lease))
+	log.Printf("Releasing lease for %q", s.leases[0].ResourceType)
```
Comment (stevekuznetsov) [inline pkg/steps/lease.go:102]: Why only print the first one?

Comment (bbguimaraes) [reply pkg/steps/lease.go]: Residual intermediary code, good catch.

**Review by petr-muller — APPROVED** [review]

**Pull request merged**